### PR TITLE
Log cleanup

### PR DESCRIPTION
**Pull request by denravonska**
Log messages cleanup:
- Remove stray newlines
- Print as integers instead of doubles where applicable
- Remove c_str() in log messages
- ~~Use log categories in miner, beacon and backup~~

### CODE REVIEW

**Comment by sedited** [comment]
Lol, I just wrote the same thing  - minus the categories.

File: src/util.cpp [hunk 1]
```diff
@@ -268,7 +268,7 @@ void LogPrintStr(const std::string &str)
 
             // Debug print useful for profiling
             if (fLogTimestamps && fStartedNewLine)
-                fprintf(fileout, "%s ", DateTimeStrFormat("%x %H:%M:%S",  GetAdjustedTime()).c_str());
+                fprintf(fileout, "%s\n", DateTimeStrFormat("%x %H:%M:%S",  GetAdjustedTime()).c_str());
```
Comment (sedited) [inline src/util.cpp]: The lines are now formatted as 
```
    time 
    message
````
I much rather prefer 
````
    time message
````

Comment (denravonska) [reply src/util.cpp]: Ah, good catch. I'll change it.

Comment (denravonska) [reply src/util.cpp]: Log timestamps would be a good thing to have enabled by default.

**Review by tomasbrod — COMMENTED** [review]
I like it from the code perspective, but from the functionality vantage, it now requires -debug=miner in addition to -debug=true to show messages. Plus there are a lot of log prints without a topic. What about changing the _important_ messages to use null category?

**Comment by denravonska** [comment]
Oh, I had no idea it worked like that, which is rather silly TBH. Then we should not use categories with the implementation we have today. I feel like our logging is lacking in the level department and would benefit if we could use categories _and_ log levels.

**Comment by tomasbrod** [comment]
Category can be used as a level, eg: `miner`, `miner2`.

**Comment by denravonska** [comment]
Yeah, maybe. I'd prefer the levels in the calls so it's more explicit.

```
Log.i("processor", "Processing block");
Log.d("processor", "Double check payments");
if(!block.ValidatePayments()) {
   Log.e("processor", "Error validating payments");
   return;
}
```

where a certain level (usually info) is enabled by default and higher levels can be enabled either globally or per category.

Until that is sorted out I will remove the category commits.

**Comment by denravonska** [comment]
@tomasbrod That sounds very sane and makes it easier to use rather than having to manually specify the category each time, and code can be moved around without disrupting the log output. So you usually have something like:

```
namespace {
   static Logger log("miner");
}

void doSomething() {
   log.Info("Doing something");
}
```
?

I like that approach.

**Comment by iFoggz** [comment]
i agree with that approach its more of a categorized logging; bitcoin does similar as well. they use a enum for log flags thou.

example of a log line by them:

LogPrint(BCLog::BENCH, "CreateNewBlock() packages: %.2fms (%d packages, %d updated descendants), validity: %.2fms (total %.2fms)\n", 0.001 * (nTime1 - nTimeStart), nPackagesSelected, nDescendantsUpdated, 0.001 * (nTime2 - nTime1), 0.001 * (nTime2 - nTimeStart));

your idea would make it as a whole however we do have some areas where we do different kinds of logging so one namespace may not work for everyone of them.

we should move to remove the debug1-5, and 10 for categorized logging of some sort

**Review by tomasbrod — APPROVED** [review]
I do not have objections to this changeset.

**Pull request merged**